### PR TITLE
Integrate included filters

### PR DESCRIPTION
**Pull request by carlthuringer**
@hatchloyalty/core 

Append our fork of the project with a cool feature rescued from the depths of JR pull requests from 2016. Tests work, and I test-integrated it with core service and it provides the desired functionality and more. Pretty spiffy.

### CODE REVIEW

File: lib/jsonapi/resource.rb [hunk 1]
```diff
@@ -551,7 +551,7 @@ def has_one(*attrs)
       end
 
       def belongs_to(*attrs)
-        ActiveSupport::Deprecation.warn "In #{name} you exposed a `has_one` relationship "\
```
Comment (tshuck) [inline lib/jsonapi/resource.rb]: Nit:  Should probably keep.

**Review by tshuck — APPROVED** [review]
Looks good and should be super helpful.

There's some spacing and stuff that rubocop would care about.  Only nit of real value is that we inadvertently removed the warning about a resource `belongs_to` and left a floating multiline string in `resource.rb`.

File: lib/jsonapi/resource.rb (hunk 1 above)
Comment (carlthuringer) [reply lib/jsonapi/resource.rb]: Dangit I restored that a couple of times but through rebasing and refactoring it snuck back in. I was as baffled as you probably were that this line was deleted. The PR was full of bizarre deletions and additions and dead-ends.

**Comment by carlthuringer** [comment]
Yeah, I don't want to rubocop this repo unless we're going to take it over. And if we do that there's `0.10` which is a significant refactoring we need to consider either moving up to, or abandoning.

**Pull request merged**